### ISSUE TITLE
 Switch Device without connection on Google Home app

### ISSUE THREAD

**Issue by JosepFarreLlados** [mikejac/node-red-contrib-google-smarthome#377]
Hello!!
I have node-red v3.0.2 and I use "node-red-contrib-google-smarthome" version 0.4.3

In node-red I setup a Switch node and I can Send spoken notifications (on/off) using a [Lenovo Smart clock](https://www.lenovo.com/es/es/smart-clock/?orgRef=https%253A%252F%252Fwww.google.com%252F) and works fine.

On other hand on [Google Home APP](https://play.google.com/store/apps/details?id=com.google.android.apps.chromecast.app&hl=ca&gl=US) of my Android phone, appear the switch button with the status "without connection". 
Google Home APP detects all nodes (Switch, Blinds...) but the  status is "without connection". Any idea why Google Home APP can't connect to the nodes?
Any help would be appreciated. 
Thank you.
Best regards,

**Comment by Caprico85** [comment]
Hi!

"without connection" means the devices are shown like this?

![image](https://user-images.githubusercontent.com/2081806/224484808-05e61683-19eb-4616-aa2a-f1c5fc4becb6.png)

So far, I have no idea what the reason could be. It can't be a general connection problem. Otherwise voice commands wouldn't work. So the problem must be something else.

Any errors on Google Cloud logs https://console.cloud.google.com/logs/? You may need to change the timespan to see the log messages (if there are any messages at all).

Do you see any error messages on Node-RED's debug panel?

Can you enable debug output in the management node and then look at Node-RED's log file. Any errors there? Most interesting part is probably when you swipe down in the app to refresh. There should be a bunch of messages when Google queries the new state of your devices.

Did you setup local fulfillment? If so, try to disable it.

![image](https://user-images.githubusercontent.com/2081806/224513083-c6069205-dba7-434b-bc61-116f5854f2ce.png)

Regards
Caprico

**Comment by JosepFarreLlados** [comment]
Hello Caprico,
Many thanks for your reply!
On Google Home APP
<https://play.google.com/store/apps/details?id=com.google.android.apps.chromecast.app&hl=ca&gl=US>
the  Switch node looks like as you shown.
I have had local fulfillment disabled.

I enable debug output in the management node and Node-RED's show:
[image: image.png]



Google Cloud logs are two:
first:
[image: image.png]
second:
[image: image.png]

Thanks in advance.
Best regards,




Missatge de Caprico ***@***.***> del dia dg., 12 de març 2023
a les 0:23:

**Comment by JosepFarreLlados** [comment]
Hello!
During these days I have observed that very few times Google Home APP worked, randomly, and when it restarts fail again.

My google console Settings, according with the [issue #332](https://github.com/mikejac/node-red-contrib-google-smarthome/issues/332), has as the default region Spain:
![Sin título](https://user-images.githubusercontent.com/98636450/224656997-e762dded-a22e-41af-b69b-9481e3e993cd.png)

Up to now the firewall only allowed the access from US and Spain. I did a change to allow access from all countries and now works.

Seems that default region/zone depends on the client, due to my Lenovo Smart clock always works and my Android phone no. Any feedback on this behavior?

Thanks in advance.
Best regards
